packages/code-block/partials/clipboard-button/index.tsx

```diff
@@ -73,9 +77,29 @@ function ClipboardButton({
   // reset to the default appearance so that it's clear
   // the "Copy" button can be used again
   useEffect(() => {
+    /**
+     * This button loses focus when re-rendered on `copiedState` change. This
+     * block brings the button back into focus if the current
+     * `document.activeElement` is `document.body` and the button has not
+     * already been re-focused. If the user has activated the button and then
+     * immediately navigated away via keyboard, the check on
+     * `document.activeElement` will prevent the user's focus from being moved
+     * back to the copy button.
+     */
+    if (
+      (copiedState === true || copiedState === false) &&
+      buttonRef &&
+      buttonRef.current &&
+      document.activeElement === document.body &&
+      document.activeElement !== buttonRef.current
+    ) {
+      buttonRef.current.focus()
+    }
```

Comment:
Do we know why the button is losing focus? Would it work to attempt to re-focus in our `onClick` handler instead? If it's happening as a side-effect of the copy functionality, it seems like that would be a better spot.

Reply:
Yes, the button is losing focus because it's getting re-rendered when `setCopiedState` is called. Let me see if calling `.focus` is possible right after `setCopiedState(true)` and `setCopiedState(false)`.

Reply:
@BRKalow I added an `updateCopiedState` helper in https://github.com/hashicorp/react-components/pull/605/commits/866849cec9cb466ab12e5e7c6c6304802d953084. How does it look?

```tsx
/**
 * Handle setting state, re-focusing the button, and invoking the copy
 * callback. This should not be called when `copiedState` is being reset, only
 * when there is a success or failure of the copy action.
 */
function updateCopiedState(newState: boolean) {
  // Update the local state variable
  setCopiedState(newState)

  // Re-focus the copy button
  buttonRef?.current?.focus()

  // If an onCopyCallback was provided
  if (typeof onCopyCallback == 'function') {
    onCopyCallback(newState)
  }
}
```